backends/p4tools/modules/testgen/core/small_step/expr_stepper.cpp

```diff
@@ -255,6 +299,13 @@ bool ExprStepper::preorder(const IR::Mux *mux) {
 
 bool ExprStepper::preorder(const IR::PathExpression *pathExpression) {
     logStep(pathExpression);
+    // If the path expression is a Type_MatchKind, convert it to a StringLiteral.
```

Comment:
Technically, it is possible to have a `Type_MatchKind` as an input but they are not declared in the namespace. The resolver will try to lookup the PathExpression and fail. StringLiterals are not resolved.

Reply:
Maybe this should be a compiler pass. But this case is extremely rare, there is only one P4 program that has this issue (https://github.com/p4lang/p4c/blob/main/testdata/p4_16_samples/issue3091.p4#L230).

Reply:
I understand and this seems like a very edge case. However, this does mean we lose the type information and therefore information that it was a match kind. Maybe it would be better to handle this type specifically when resolving? Or maybe create a new wrapper node (e.g. `UnresolvedExpression` (containing `IR::Expression`), or `UnresolvedPathExpression` (containing `IR::Path`)) and use it for both this and the block names and not resolve that. That would allow not resolving these yet it would preserve the type information.

Reply:
We are preserving the `Type_MatchKind` by setting the type of the StringLiteral to `Type_MatchKind`. This way we are still able to distinguish. In fact, the interpreter would throw an error if the type of the resolved expression does not match with the original expression.

I think the StringLiteral does serve this use case better than creating a new type, as `UnresolvedExpression` would ultimately have the same properties. 

We also simply could not support this use-case until we know exactly what to do with an input argument that is a match kind.

Reply:
Oh, I did not realize a string literal has a type. Somehow I automatically expected it can only be of a string type. In that case, I guess we can use the string literal.